ipn/localapi/localapi.go

```diff
@@ -1106,7 +1108,7 @@ func (h *Handler) serveFiles(w http.ResponseWriter, r *http.Request) {
 				http.Error(w, "invalid waitsec", http.StatusBadRequest)
 				return
 			}
-			deadline := time.Now().Add(time.Duration(d) * time.Second)
+			deadline := h.clock.Now().Add(time.Duration(d) * time.Second)
 			var cancel context.CancelFunc
 			ctx, cancel = context.WithDeadline(ctx, deadline)
```

Comment:
This'll be slightly awkward until `tstime.Clock` also has a way to implement `context.WithDeadline` against simulated time.

Reply:
added followup issue here https://github.com/tailscale/tailscale/issues/8604